doc/zephyr.doxyfile.in

```diff
@@ -807,7 +807,9 @@ RECURSIVE              = YES
 EXCLUDE                = @ZEPHYR_BASE@/include/spi_legacy.h \
                          @ZEPHYR_BASE@/include/net/http_legacy.h \
                          @ZEPHYR_BASE@/include/misc/util.h \
-                         @ZEPHYR_BASE@/include/cmsis_rtos_v1/cmsis_os.h
+                         @ZEPHYR_BASE@/include/cmsis_rtos_v1/cmsis_os.h \
+                         @ZEPHYR_BASE@/include/net/coap.h \
```

Comment:
This is the real "trick" for fixing the duplicate names errors (since coap.h and coap_sock.h define conflicting names).